src/sentry/integrations/slack/notify_action.py

```diff
@@ -121,13 +131,18 @@ def after(self, event, state):
 
         def send_notification(event, futures):
             rules = [f.rule for f in futures]
-            attachment = build_group_attachment(event.group, event=event, tags=tags, rules=rules)
+            attachments = [build_group_attachment(event.group, event=event, tags=tags, rules=rules)]
+            # check if we should have the upgrade notice attachment
+            if get_integration_type(integration) == "workspace_app":
+                if random.random() < UPGRADE_MESSAGE_FREQUENCY:
```

Comment:
Nit: Not sure if it's more pythonic to do nested `if` logic or flatten with `and`.
```suggestion
if get_integration_type(integration) == "workspace_app" and random.random() < UPGRADE_MESSAGE_FREQUENCY:

Reply:
@NisanthanNanthakumar I purposely did it this way for readability so we don't have the line wrapping